serenity/data_utils.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef SERENITY_DATA_UTILS_HPP
+#define SERENITY_DATA_UTILS_HPP
+
+#include "serenity/metrics_helper.hpp"
+#include "serenity/serenity.hpp"
+
+namespace mesos {
+namespace serenity {
+namespace usage {
+
+//! Resource Usage getters.
+using GetterFunction = Try<double_t>(
+    const ResourceUsage_Executor& previousExec,
+    const ResourceUsage_Executor& currentExec);
+
+
+inline Try<double_t> getIpc(
+    const ResourceUsage_Executor& previousExec,
+    const ResourceUsage_Executor& currentExec) {
+  Try<double_t> ipc = CountIpc(previousExec, currentExec);
+  if (ipc.isError()) return Error(ipc.error());
+
+  return ipc;
+}
+
+
+inline Try<double_t> getEmaIpc(
+    const ResourceUsage_Executor& previousExec,
+    const ResourceUsage_Executor& currentExec) {
+  if (!currentExec.statistics().has_net_tcp_active_connections())
```

Comment:
Because we have to save this double value (EMA of IPC) into some field in ResourceUsage. We choose not used "net tcp connections". As you remember we choose to use ResourceUsage between filters instead of creating seperate protos for all data tranformations. Of course we can change it in future, because that method is quite ambigous.